.github/workflows/main.yml

```diff
@@ -26,10 +26,212 @@ env:
+        docker build --cache-from=${IMG_2404} -t "${{ steps.docker_img.outputs.img_name }}" - < tools/docker/24.04/Dockerfile
+        docker push "${{ steps.docker_img.outputs.img_name }}"
+
+    outputs:
+      docker_img_name: "${{ steps.docker_img.outputs.img_name }}"
+
+  # Build a basic version of kotekan
+  build-base-2404:
+    runs-on: ubuntu-latest
+    needs: build-docker-2404
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v3
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Cache ccache files
+      uses: actions/cache@v4
+      with:
+        path: .ccache
+        key: ccache-base-build-2404-${{ github.sha }}
+        restore-keys: |
+          ccache-base-build-2404
+          ccache-full-build-2404
+
+    - name: Build kotekan
+      run: |
+        OPTS=(--rm --mount type=bind,src=$(pwd),target=/code/kotekan -w /code/kotekan/build-2404/ ${{ needs.build-docker-2404.outputs.docker_img_name }})
+        docker run "${OPTS[@]}" ccache -s
+        docker run "${OPTS[@]}" \
+          cmake -Wdev -Werror=dev -Wdeprecated -Werror=deprecated \
+            -DWERROR=ON \
+            -DCMAKE_LINK_WHAT_YOU_USE=ON \
+            -DCMAKE_BUILD_TYPE=Test \
+            -DCCACHE=ON \
+            -DWITH_TESTS=ON ..
+        docker run "${OPTS[@]}" make -j 2
+        docker run "${OPTS[@]}" ccache -s
+
+  # Build a full version of kotekan and run the unit tests
+  build-full-tests-2404:
+    runs-on: ubuntu-latest
+    needs: build-docker-2404
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v3
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Cache ccache files
+      uses: actions/cache@v4
+      with:
+        path: .ccache
+        key: ccache-full-build-2404-${{ github.sha }}
+        restore-keys: |
+          ccache-full-build-2404
+          ccache-base-build-2404
+
+    - name: Free disk space
+      if: ${{ !env.ACT }} # Don't free up if running locally
+      run: |
+          df -h
+          docker rmi $(docker image ls -aq)
+          df -h
+          sudo apt-get clean
+          sudo apt-get autoclean
+          df -h
+
+    - name: Build kotekan
+      run: |
+        OPTS=(--rm --mount type=bind,src=$(pwd),target=/code/kotekan -w /code/kotekan/build-2404/ ${{ needs.build-docker-2404.outputs.docker_img_name }})
+        docker run "${OPTS[@]}" ccache -s
+        docker run "${OPTS[@]}" \
+          cmake -Wdev -Werror=dev -Wdeprecated -Werror=deprecated \
+            -DWERROR=ON \
+            -DCMAKE_LINK_WHAT_YOU_USE=ON \
+            -DCMAKE_BUILD_TYPE=Test \
+            -DUSE_ASDF=ON \

```

Comment:
I think we should include `USE_CUDA` and `USE_OPENCL` here, since otherwise we could miss issues with the GPU related compile in 24.04.   Or we should have a separate build like in the 22.04 case for the GPU code.

Reply:
Agreed - though, I was thinking to add these tests in a future PR?

Reply:
Added!